clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2SecurityGroup.java

```diff
@@ -119,7 +120,13 @@ public static KubernetesV2SecurityGroup fromCacheData(CacheData cd) {
       return Collections.emptySet();
     }
     return policy.getSpec().getIngress().stream()
-        .map(i -> i.getPorts().stream().map(KubernetesV2SecurityGroup::fromPolicyPort))
+        .map(
```

Comment:
The original code was nesting streams, which made things hard to read (possibly contributing to the bug).  It would be great to clean this up into a single stream along with this fix.  Something like the following should be equivalent and also fix the NPE:
```
    return policy.getSpec().getIngress().stream()
      .map(V1NetworkPolicyIngressRule::getPorts)
      .filter(Objects::nonNull)
      .flatMap(Collection::stream)
      .map(KubernetesV2SecurityGroup::fromPolicyPort)
      .collect(Collectors.toSet());
```
Do you mind changing this (and the stream below) to use that single-stream style?

Reply:
Not at all, that version is much nicer. Changed with 2757398ebece6c43f22034c53f727c2ea8cb8469